packages/effect/src/Array.ts

```diff
@@ -1097,6 +1097,8 @@ export const replaceOption: {
   <A, B>(self: Iterable<A>, i: number, b: B): Option<Array<A | B>> => modifyOption(self, i, () => b)
 )
 
+type TypeOfIterable<T extends Iterable<any>> = T extends Iterable<infer A> ? A : never
```

Comment:
This is how I fixed the inference of `a`. Tell me if:
- it's a valid way to fix the issue
- it's the correct place to put this kind of the type helpers (or if it already exists somewhere. I did not find anything on my own)